src/mscorlib/src/Microsoft/Win32/RegistryKey.cs

```
@@ -232,77 +137,44 @@ public void DeleteValue(String name, bool throwOnMissingValue)
         }
 
-        internal static RegistryKey GetBaseKey(IntPtr hKey, RegistryView view)
+        private static RegistryKey GetBaseKey(IntPtr hKey, string keyName)
         {
-            int index = ((int)hKey) & 0x0FFFFFFF;
-            BCLDebug.Assert(index >= 0 && index < hkeyNames.Length, "index is out of range!");
-            BCLDebug.Assert((((int)hKey) & 0xFFFFFFF0) == 0x80000000, "Invalid hkey value!");
-
-            bool isPerf = hKey == HKEY_PERFORMANCE_DATA;
-            // only mark the SafeHandle as ownsHandle if the key is HKEY_PERFORMANCE_DATA.
-            SafeRegistryHandle srh = new SafeRegistryHandle(hKey, isPerf);
+            SafeRegistryHandle srh = new SafeRegistryHandle(hKey, ownsHandle: false);
 
-            RegistryKey key = new RegistryKey(srh, true, true, false, isPerf, view);
-            key.checkMode = RegistryKeyPermissionCheck.Default;
-            key.keyName = hkeyNames[index];
+            RegistryKey key = new RegistryKey(srh, true, true);
+            key.keyName = keyName;
             return key;
         }
 
-        /**
-         * Retrieves a subkey. If readonly is <b>true</b>, then the subkey is opened with
-         * read-only access.
-         *
-         * @param name Name or path of subkey to open.
-         * @param readonly Set to <b>true</b> if you only need readonly access.
-         *
-         * @return the Subkey requested, or <b>null</b> if the operation failed.
-         */
+        /// <summary>
+        /// Retrieves a subkey or null if the operation failed.
+        /// </summary>
+        /// <param name="writable">True to open writable, otherwise opens the key read-only.</param>
         public RegistryKey OpenSubKey(string name, bool writable)
         {
             ValidateKeyName(name);
             EnsureNotDisposed();
-            name = FixupName(name); // Fixup multiple slashes to a single slash
```

As it is internal only for this copy of the code I felt it was better to fail hard here and cut out a ton of code. This just allows you to specify keys with double separators and ending separators.